src/qos_client/src/cli/services.rs

```diff
@@ -1530,6 +1530,18 @@ pub(crate) fn p256_asymmetric_decrypt<P: AsRef<Path>>(
 	Ok(())
 }
 
+pub(crate) fn extract_and_write_ephemeral_key_hex<P: AsRef<Path>>(
+	attestation_doc_path: P,
+	ephemeral_key_path: P,
+) -> Result<(), Error> {
+	let bytes =
+		fs::read(attestation_doc_path).map_err(|e| Error::ReadShare(e.to_string()))?;
+	let attestation_doc: AttestationDoc = extract_attestation_doc(bytes.as_ref(), true, None);
+	let ephemeral_key: P256Public = P256Public::from_bytes(&attestation_doc.public_key.expect("No ephemeral key in the attestation doc"),).expect("Ephemeral key not valid public key");
```

Comment:
Just a note for the future if anyone ever wants to cleanup the CLI - the code is super inconsistent on when it chooses to panic vs bubble up an error. The expects are easier to write but probably better practice to bubble up an error. The issue is we don't use any error handling libraries so the error enum stuff is very unergonomic. If we didn't care about supply chain stuff I would suggest using anyhow or eyre. They are super popular and allow you to mix declaring error messages in place and using enums

https://docs.rs/anyhow/latest/anyhow/
https://docs.rs/eyre/latest/eyre/